lib/CfTemplateGenerators/Lambda.js

```diff
@@ -13,10 +13,12 @@ function buildUpdatePolicy ({ codeDeployApp, deploymentGroup, afterHook, beforeH
   return omitEmpty({ UpdatePolicy: updatePolicy })
 }
 
-function buildAlias ({ alias, functionName, functionVersion, trafficShiftingSettings }) {
+function buildAlias ({ alias, functionName, functionVersion, trafficShiftingSettings, targetAliasTemplate }) {
   const lambdaAlias = {
+    ...targetAliasTemplate,
     Type: 'AWS::Lambda::Alias',
     Properties: {
+      ...targetAliasTemplate ? targetAliasTemplate.Properties : null,
```

Comment:
JS is the worst. This had me confused for so long, but I think (logically) this is doing something like: 

```
!!targetAliasTemplate ? ...targetAliasTemplate.Properties : null
```

... in fact, would the above achieve the same thing? If so, I think it's perhaps slightly neater. I was originally confused as I thought the conditional check was being done on `...targetAliasTemplate` but I think the `...` is being applied to the result of the ternary condition?

Reply:
Oh JS, please never change (but actually PLEASE CHANGE IMMEDIATELY!) 

<img width="316" alt="image" src="https://github.com/Flagsmith/serverless-plugin-canary-deployments/assets/14089968/98dbef63-069f-496c-9c3c-21b342a9e954">

(there is no action here, this is just a rant)

Reply:
Added parentheses for hopefully less confusion.